File: schemas/modelschemas/deployment_revision.go
package modelschemas

type DeploymentRevisionStatus string

const (
	DeploymentRevisionStatusActive   DeploymentRevisionStatus = "active"
	DeploymentRevisionStatusInactive DeploymentRevisionStatus = "inactive"
)

func (d DeploymentRevisionStatus) Ptr() *DeploymentRevisionStatus {
	return &d
}

func DeploymentRevisionStatusPtr(status DeploymentRevisionStatus) *DeploymentRevisionStatus {
	return status.Ptr()
}

File: dashboard/src/schemas/deployment_revision.ts
/* eslint-disable import/no-cycle */
import { IDeploymentTargetSchema } from './deployment_target'
import { IUserSchema } from './user'
import { IResourceSchema } from './resource'

export type DeploymentRevisionStatus = 'active' | 'inactive'

export interface IDeploymentRevisionSchema extends IResourceSchema {
    creator?: IUserSchema
    status: DeploymentRevisionStatus
    targets: IDeploymentTargetSchema[]
}
